File: fix_ui.js
#!/usr/bin/env node
/**
 * Fix script for Discord Manager UI issues.
 * Run this from the discord-manager directory: node fix_ui.js
 */
const http = require('http');
const fs = require('fs');
const path = require('path');
const { spawn } = require('child_process');

const BASE = __dirname;

function fetch(url) {
    return new Promise((resolve, reject) => {
        http.get(url, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => resolve({ status: res.statusCode, data }));
        }).on('error', reject);
    });
}

async function checkBridge() {
    try {
        const r = await fetch('http://127.0.0.1:8787/i18n/languages');
        if (r.status === 200) {
            const data = JSON.parse(r.data);
            console.log(`✓ /i18n/languages OK: ${data.languages?.length || 0} languages`);
            return true;
        } else {
            console.log(`✗ /i18n/languages returned ${r.status}`);
            return false;
        }
    } catch (e) {
        console.log(`✗ Bridge not reachable: ${e.message}`);
        return false;
    }
}

async function checkLocale(code) {
    try {
        const r = await fetch(`http://127.0.0.1:8787/i18n/locales/${code}`);
        if (r.status === 200) {
            const data = JSON.parse(r.data);
            const keys = Object.keys(data.strings || {}).length;
            console.log(`✓ /i18n/locales/${code} OK: ${keys} translation keys`);
            return true;
        } else {
            console.log(`✗ /i18n/locales/${code} returned ${r.status}`);
            return false;
        }
    } catch (e) {
        console.log(`✗ Locale fetch failed: ${e.message}`);
        return false;
    }
}

async function checkStatic(file) {
    try {
        const r = await fetch(`http://127.0.0.1:8787/${file}`);
        if (r.status === 200) {
            console.log(`✓ /${file} OK (${r.data.length} bytes)`);
            return true;
        } else {
            console.log(`✗ /${file} returned ${r.status}`);
            return false;
        }
    } catch (e) {
        console.log(`✗ Static file fetch failed: ${e.message}`);
        return false;
    }
}

function clearElectronCache() {
    const dirs = [
        process.env.APPDATA ? path.join(process.env.APPDATA, 'Discord Manager') : null,
        process.env.HOME ? path.join(process.env.HOME, 'Library', 'Application Support', 'Discord Manager') : null,
        process.env.HOME ? path.join(process.env.HOME, '.config', 'Discord Manager') : null,
    ].filter(Boolean);
    
    for (const d of dirs) {
        if (fs.existsSync(d)) {
            console.log(`Removing cache: ${d}`);
            fs.rmSync(d, { recursive: true, force: true });
        }
    }
}

function verifyWwwFiles() {
    const required = ['index.html', 'app.js', 'styles.css', 'locales/en.json', 'locales/zh.json', 'locales/it.json'];
    for (const f of required) {
        const p = path.join(BASE, 'www', f);
        if (fs.existsSync(p)) {
            console.log(`  ✓ www/${f}`);
        } else {
            console.log(`  ✗ MISSING: www/${f}`);
        }
    }
}

function checkPackageJson() {
    const pkg = JSON.parse(fs.readFileSync(path.join(BASE, 'package.json'), 'utf8'));
    const asarUnpack = pkg.build?.asarUnpack || [];
    if (asarUnpack.includes('www/locales/**/*')) {
        console.log('  ✓ asarUnpack includes www/locales/**/*');
    } else {
        console.log('  ✗ asarUnpack MISSING www/locales/**/*');
    }
}

async function main() {
    console.log('='.repeat(60));
    console.log('Discord Manager UI Fix Script');
    console.log('='.repeat(60));
    
    // 1. Clear Electron cache
    console.log('\n1. Clearing Electron cache...');
    clearElectronCache();
    
    // 2. Check if bridge is running
    console.log('\n2. Checking bridge endpoints...');
    const bridgeOk = await checkBridge();
    
    if (bridgeOk) {
        await checkLocale('en');
        await checkLocale('zh');
        await checkLocale('it');
        await checkStatic('styles.css');
        await checkStatic('app.js');
    } else {
        console.log('\nBridge not running. Starting it...');
        const env = { ...process.env };
        env.WWW_DIR = path.join(BASE, 'www');
        env.DATA_DIR = path.join(BASE, 'data', 'messages');
        env.VOICE_DIR = path.join(BASE, 'data', 'voice');
        env.LOCALES_DIR = path.join(BASE, 'www', 'locales');
        
        const proc = spawn('node', ['bridge.js'], { cwd: BASE, env, stdio: 'inherit' });
        console.log(`Bridge started (PID: ${proc.pid}), waiting 3s...`);
        await new Promise(r => setTimeout(r, 3000));
        
        if (await checkBridge()) {
            await checkLocale('en');
            await checkStatic('styles.css');
            await checkStatic('app.js');
        } else {
            console.log('ERROR: Bridge failed to start. Check console for errors.');
            proc.kill();
            process.exit(1);
        }
    }
    
    // 3. Verify www files
    console.log('\n3. Verifying www/ files...');
    verifyWwwFiles();
    
    // 4. Check package.json
    console.log('\n4. Checking package.json...');
    checkPackageJson();
    
    console.log('\n' + '='.repeat(60));
    console.log('NEXT STEPS:');
    console.log('='.repeat(60));
    console.log('If all checks passed above, run:');
    console.log('  npm run dev');
    console.log('');
    console.log('If bridge failed, run manually to see errors:');
    console.log('  node bridge.js');
    console.log('');
    console.log('Common issues:');
    console.log('  - Missing ws module: npm install ws');
    console.log('  - Port 8787 in use: kill other node processes');
    console.log('  - Locale files not found: check www/locales/ exists');
}

main().catch(console.error);

File: fix_ui.py
#!/usr/bin/env python3
"""
Fix script for Discord Manager UI issues.
Run this from the discord-manager directory.
"""
import subprocess
import sys
import os
import time
import requests
from pathlib import Path

BASE = Path(__file__).parent

def run_cmd(cmd, cwd=None, timeout=30):
    """Run command and return (success, output)"""
    try:
        result = subprocess.run(cmd, cwd=cwd or BASE, capture_output=True, text=True, timeout=timeout, shell=True)
        return result.returncode == 0, result.stdout + result.stderr
    except subprocess.TimeoutExpired:
        return False, "Timeout"
    except Exception as e:
        return False, str(e)

def check_bridge():
    """Check if bridge is running and endpoints work"""
    try:
        r = requests.get("http://127.0.0.1:8787/i18n/languages", timeout=3)
        if r.status_code == 200:
            data = r.json()
            print(f"✓ /i18n/languages OK: {len(data.get('languages', []))} languages")
            return True
        else:
            print(f"✗ /i18n/languages returned {r.status_code}")
            return False
    except Exception as e:
        print(f"✗ Bridge not reachable: {e}")
        return False

def check_locale(code='en'):
    try:
        r = requests.get(f"http://127.0.0.1:8787/i18n/locales/{code}", timeout=3)
        if r.status_code == 200:
            data = r.json()
            keys = len(data.get('strings', {}))
            print(f"✓ /i18n/locales/{code} OK: {keys} translation keys")
            return True
        else:
            print(f"✗ /i18n/locales/{code} returned {r.status_code}")
            return False
    except Exception as e:
        print(f"✗ Locale fetch failed: {e}")
        return False

def check_static_file(path):
    try:
        r = requests.get(f"http://127.0.0.1:8787/{path}", timeout=3)
        if r.status_code == 200:
            print(f"✓ /{path} OK ({len(r.content)} bytes)")
            return True
        else:
            print(f"✗ /{path} returned {r.status_code}")
            return False
    except Exception as e:
        print(f"✗ Static file fetch failed: {e}")
        return False

def clear_electron_cache():
    """Clear Electron user data cache"""
    import shutil
    cache_dirs = [
        Path(os.environ.get('APPDATA', '')) / 'Discord Manager',
        Path.home() / 'Library' / 'Application Support' / 'Discord Manager',
        Path.home() / '.config' / 'Discord Manager',
    ]
    for d in cache_dirs:
        if d.exists():
            print(f"Removing cache: {d}")
            shutil.rmtree(d, ignore_errors=True)

def main():
    print("=" * 60)
    print("Discord Manager UI Fix Script")
    print("=" * 60)
    
    # 1. Check if we're in the right directory
    if not (BASE / 'bridge.js').exists():
        print(f"ERROR: Run this from discord-manager directory")
        sys.exit(1)
    
    # 2. Clear Electron cache (fixes stale UI)
    print("\n1. Clearing Electron cache...")
    clear_electron_cache()
    
    # 3. Check if bridge is running
    print("\n2. Checking bridge endpoints...")
    if check_bridge():
        check_locale('en')
        check_locale('zh')
        check_locale('it')
        check_static_file('styles.css')
        check_static_file('app.js')
    else:
        print("\nBridge not running. Starting it...")
        # Start bridge in background
        env = os.environ.copy()
        env['WWW_DIR'] = str(BASE / 'www')
        env['DATA_DIR'] = str(BASE / 'data' / 'messages')
        env['VOICE_DIR'] = str(BASE / 'data' / 'voice')
        env['LOCALES_DIR'] = str(BASE / 'www' / 'locales')
        
        proc = subprocess.Popen(['node', 'bridge.js'], cwd=BASE, env=env)
        print(f"Bridge started (PID: {proc.pid}), waiting 3s...")
        time.sleep(3)
        
        if check_bridge():
            check_locale('en')
            check_static_file('styles.css')
            check_static_file('app.js')
        else:
            print("ERROR: Bridge failed to start. Check console for errors.")
            proc.terminate()
            sys.exit(1)
    
    # 4. Verify www files exist
    print("\n3. Verifying www/ files...")
    required = ['index.html', 'app.js', 'styles.css', 'locales/en.json', 'locales/zh.json', 'locales/it.json']
    for f in required:
        p = BASE / 'www' / f
        if p.exists():
            print(f"  ✓ www/{f}")
        else:
            print(f"  ✗ MISSING: www/{f}")
    
    # 5. Check package.json has asarUnpack
    print("\n4. Checking package.json...")
    import json
    with open(BASE / 'package.json') as f:
        pkg = json.load(f)
    asar_unpack = pkg.get('build', {}).get('asarUnpack', [])
    if 'www/locales/**/*' in asar_unpack:
        print("  ✓ asarUnpack includes www/locales/**/*")
    else:
        print("  ✗ asarUnpack MISSING www/locales/**/*")
    
    print("\n" + "=" * 60)
    print("NEXT STEPS:")
    print("=" * 60)
    print("If all checks passed above, run:")
    print("  npm run dev")
    print()
    print("If bridge failed, check terminal for errors when running:")
    print("  node bridge.js")
    print()
    print("Common issues:")
    print("  - Missing 'ws' module: npm install ws")
    print("  - Port 8787 in use: kill other node processes")
    print("  - Locale files not found: check www/locales/ exists")

if __name__ == '__main__':
    main()